cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/BasicPushStrategy.java

```diff
@@ -90,12 +90,21 @@
 
     public PushAttachmentsInline pushAttachmentsInline = PushAttachmentsInline.Small;
 
+    // request and response interceptors are only held here for convenience - they are eventually
+    // used by the underlying CouchClient instance
+
+    protected List<HttpConnectionRequestInterceptor> requestInterceptors;
+
+    protected List<HttpConnectionResponseInterceptor> responseInterceptors;
+
```

Comment:
Right now it's only for the benefit of the test cases. I wasn't entirely happy with this. Some alternatives are: widening the access to the fields on the CouchClient, using reflection in the test cases, not doing the asserts on the interceptors.